cps2pc

```diff
@@ -0,0 +1,206 @@
+    elif kind == "jar":
+        kind_dict['classpath'] = location
+    elif kind == "interface":
+        pass
+    else:
+        raise Exception('unknown')
+    return kind_dict
+
+def write_pc(pc_dict, filename, prefix):
+    def list_to_str(l):
+        return " ".join(l)
+    convert = {'unicode': unicode, 'str': str, 'list': list_to_str}
+    with open(filename, "w") as f:
+        for k,v in pc_dict.iteritems():
+            f.write("%s: %s\n" % (k, convert[type(v).__name__](v).replace('@prefix@', prefix)))
+
+def remove_empty_keys(pc_dict):
+    for k, v in pc_dict.iteritems():
+        if not v:
+            pc_dict.pop(k)
+
+def optional(extensions, attr):
+    try:
+        return extensions[attr]
+    except:
+        return []
+
+#------------------------------------------------------------------------------
+def write_targets_config(package, output_dir, compiler, prefix):
+    platform = get_platform(package)
+    for component_name, component in package.components.iteritems():
+        # Ignored CPS tags (not supported by pkgconfig):
+        # * Link-Location
+        # * Link-Languages
+        # * Extensions
+        # * Configurations
+        # Missing pkgconfig tag (not supported by CPS):
+        # * Description
+        pc_dict = OrderedDict()
+        pc_dict['Name'] = component_name
+        # Handle supplemental schema
+        for element in ['Description', 'Website']:
+            pc_dict[element] = optional(package, element.lower())
+
+        pc_dict['Version'] = package.version
+        pc_dict['Libs'] = []
+        pc_dict['Libs'] += ["-l%s" % x for x in component.link_libraries]
+        pc_dict['Libs'] += ["-l%s" % x for x in component.link_requires]
+        pc_dict['Libs'] += component.link_flags[None]
+        pc_dict['Libs'] += get_link_features(component.link_features, platform)
+        pc_dict['Libs'] += get_cpp_runtime_vendor(package)
+        try:
+            pc_dict.update(handle_kind(component.kind, component.location))
+        except Exception as e:
+            print 'Skipping %s: %s kind is not supported by pkgconfig' % (component_name, e)
+            continue
+        pc_dict['Cflags'] = []
+        pc_dict['Cflags'] += list(component.compile_flags[None])
+        pc_dict['Cflags'] += ["-I%s" % x for x in component.includes[None]]
+        pc_dict['Cflags'] += get_definitions_flags(component.definitions[None], compiler)
+        pc_dict['Cflags'] += get_compile_features(component.compile_features, compiler)
+        pc_dict['Requires'] = [x.split(':')[-1] for x in component.requires]  # Remove possible package name
+        remove_empty_keys(pc_dict)
+        filename = os.path.join(output_dir, component_name + ".pc")
+        write_pc(pc_dict, filename, prefix)
+
+
+def check_compiler_support(compiler):
+    if compiler not in _supported_compilers:
+        raise Exception('Compiler %s not supported. Check supported compiler with --supported-compiler' % compiler)
+
+
```

Comment:
Class should have `#====` separator line before, and methods should have (indented `#----`) separator lines. Also, can the class be defined before the free functions? (If "yes", there should also be a `#%%%%` separator between classes and free functions. See [cps.py](https://github.com/mwoehlke/pycps/blob/master/cps.py), especially circa [line 172](https://github.com/mwoehlke/pycps/blob/a34ffa1f5574819224f6a7ee4fb47ed255e086d4/cps.py#L172) for an example.)

All that said, this seems like an overly complicated way to handle something that would be better (and *much* more simply) handled by [`choices`](https://docs.python.org/2/library/argparse.html#choices). Accordingly, I am going to refrain from any further comment on this class.

Reply:
True. I simplified the code.